components/game/context.tsx

```diff
@@ -65,13 +66,14 @@ export const GameContextProvider = ({
   }, [id]);
 
   const move = useCallback(
-    (coords: Coords) => {
+    (coords: Coords, currentPlayer: Player, gameType: GameType) => {
       setIsMoving(true);
       setMoveError(undefined);
+      const playerTurn = gameType === "automatic" ? currentPlayer : player;
```

Comment:
both currentPlayer and gameType are supposedly known on the backend, you can have this computation there, it'd free up a lot of extra context load I believe